JustSaying/Messaging/MessageHandling/MessageContext.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using SQSMessage = Amazon.SQS.Model.Message;
+
+namespace JustSaying.Messaging.MessageHandling
+{
+    public class MessageContext
+    {
+        public MessageContext(SQSMessage message, Uri queueUri)
+        {
+            Message = message ?? throw new ArgumentNullException(nameof(message));
+            QueueUri = queueUri ?? throw new ArgumentNullException(nameof(queueUri));
+        }
+
+        public SQSMessage Message { get; }
```

Comment:
Does there need to be an SNS equivalent for this, or are they the same thing in this context?

Reply:
That's not needed, we only read from SQS queues. 

The distinction exists only on publishing where you can publish directly to a SQS queue, or more usually, indirectly by a SNS endpoint that forwards to all attached queues,.